app/models/user_annotation.rb

```diff
@@ -366,4 +366,116 @@ def self.delete_queued_annotations
     end
   end
 
+  def persist_to_study(current_user)
```

Comment:
This method looks good, but it should have a few 'progress' logger statements so that if we encounter errors, we will have more information as to where exactly the error occurred (since we won't have a stack trace).  Particularly, you should do this right before and right after creating a new record.